### DISCUSSION TITLE
How to get the consumed CPU time in each time step?

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#27954] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hi all,

We can get the total consumed CPU time when we finish the calculation. But how do we get the total consumed CPU time in each time step? 

Thanks!

**Comment by GiudGiud** [comment]
Hello

I'm not sure we have something for this.
You ll want to create a postprocessor to report the delta between the starting CPU time and the current CPU time. or do differences between time steps. 
You should look through the framework folder first, I may have missed it. It would be in the PPs, Reporters or VPPs if it exists

Guillaume

**Reply by SomnusYu** [reply]
Hi, I have searched through the framework folder, but I can not find any type (PPS, Reporters or VPP) that can report the current CPU time.

**Reply by GiudGiud** [reply]
Ok thanks for double checking

You can look at the perf graph for this otherwise. 

The perf graph will have a lot of the relevant code to perform this measurement

**Reply by SomnusYu** [reply]
Ok, thanks!
